cmd/joy/release.go

```diff
@@ -181,3 +193,179 @@ This command requires the jac cli: https://github.com/nestoca/jac
+					return fmt.Errorf("failed to pull helm chart: %w", err)
+				}
+			}
+
+			values, err := hydrateValues(release.Spec.Values, release, environment)
+			if err != nil {
+				fmt.Println("error hydrating values:", err)
+				fmt.Println("fallback to raw release.spec.values")
+				values = release.Spec.Values
+			}
+
+			if err := helm.Render(cmd.Context(), cmd.OutOrStdout(), chartPath, values); err != nil {
+				return fmt.Errorf("failed to render chart: %w", err)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&env, "env", "e", "", "environment to select release from.")
+
+	return cmd
+}
+
+func GetEnvironment(environments []*v1alpha1.Environment, name string) (*v1alpha1.Environment, error) {
+	if name == "" {
+		return environment.SelectSingle(environments, nil, "Select environment")
+	}
+
+	selectedEnv := environment.FindByName(environments, name)
+	if selectedEnv == nil {
+		return nil, fmt.Errorf("not found: %s", name)
+	}
+
+	return selectedEnv, nil
+}
+
+func getRelease(releases []*cross.Release, name, env string) (*v1alpha1.Release, error) {
+	if name == "" {
+		return getReleaseViaPrompt(releases, env)
+	}
+
+	for _, crossRelease := range releases {
+		if crossRelease.Name != name {
+			continue
+		}
+		for _, release := range crossRelease.Releases {
+			if release == nil {
+				continue
+			}
+			if release.Environment.Name == env {
+				return release, nil
+			}
+		}
+		return nil, fmt.Errorf("not found within environment %s: %s", env, name)
+	}
+
+	return nil, fmt.Errorf("not found: %s", name)
+}
+
+func getReleaseViaPrompt(releases []*cross.Release, env string) (*v1alpha1.Release, error) {
+	var (
+		candidateNames    []string
+		candidateReleases []*v1alpha1.Release
+	)
+
+	for _, crossRelease := range releases {
+		for _, release := range crossRelease.Releases {
+			if release == nil {
+				continue
+			}
+			if release.Environment.Name == env {
+				candidateNames = append(candidateNames, release.Name)
+				candidateReleases = append(candidateReleases, release)
+				break
+			}
+		}
+	}
+
+	var idx int
+	if err := survey.AskOne(&survey.Select{Message: "Select release", Options: candidateNames, PageSize: 20}, &idx); err != nil {
+		return nil, fmt.Errorf("failed prompt: %w", err)
+	}
+
+	return candidateReleases[idx], nil
+}
+
+func hydrateValues(values map[string]any, release *v1alpha1.Release, environment *v1alpha1.Environment) (map[string]any, error) {
+	data, err := yaml.Marshal(values)
+	if err != nil {
+		return nil, err
+	}
+
+	tmpl, err := template.New("").Parse(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	var builder bytes.Buffer
+	params := struct {
+		Release     *v1alpha1.Release
+		Environment *v1alpha1.Environment
+	}{
+		release,
+		environment,
+	}
+
+	if err := tmpl.Execute(&builder, params); err != nil {
+		return nil, err
+	}
+
+	var result map[string]any
+	if err := yaml.Unmarshal(builder.Bytes(), &result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
```

Comment:
When generating the value, we need to inject the `image.tag` value from the version field in the joy spec.

https://github.com/nestoca/infra/blob/7d7daedc210199d5539d009431e94f1dc7b7ff36/tools/joy-converter/joy_converter.py#L439

Reply:
If I do this it breaks certain chart renders like redis:
```
Error: YAML parse error on redis/templates/master/application.yaml: error converting YAML to JSON: yaml: line 70: mapping values are not allowed in this context
```

I also searched the catalog and never see `image.tag` except for in `podinfo` as the spec.versionKey.
Very confused image.tag

Reply:
Yah, it's because it's specific to that chart. I am not sure if it is worth handling and just assume that the `image.tag` will be rendered incorrectly.